ext/sodium/libsodium.c

```diff
@@ -3294,15 +3171,15 @@ PHP_FUNCTION(sodium_unpad)
 		RETURN_THROWS();
 	}
 	if (blocksize <= 0) {
-		zend_throw_exception(sodium_exception_ce, "block size cannot be less than 1", 0);
+		zend_argument_error(sodium_exception_ce, 2, "to be greater than 0");
 		RETURN_THROWS();
 	}
 	if (blocksize > SIZE_MAX) {
-		zend_throw_exception(sodium_exception_ce, "block size is too large", 0);
+		zend_argument_error(sodium_exception_ce, 2, "to be less than or equal to %d", SIZE_MAX);
```

Comment:
Per your comment above, this shouldn't include SIZE_MAX.

Reply:
I fixed this, but the new message is a bit weird too. For now, it's `.. to be less than or equal to the maximum allowed value`.